pyfar/dsp/regularization.py

```diff
@@ -0,0 +1,337 @@
+"""
+Documentation for regularization class.
+"""
+
+import pyfar as pf
+from pyfar.dsp.dsp import _cross_fade
+import numpy as np
+
+class RegularizedSpectrumInversion():
+    r"""Class for frequency dependent regularization and inversion.
+
+    Regularization is used in inverse filtering methods to limit the gain
+    applied by an inverse filter. This, for example, is nescessary to
+    avoid extreme amplification caused by notches and peaks in the original
+    signal's spectrum or by the limit of a system's bandwidth.
+
+    The frequency dependent regularization :math:`\epsilon(f)` can be defined
+    using one of the class methods.
+    The regularization is then applied when calculating the inverse of a
+    signal as [1]_, [2]_:
+
+    .. math::
+
+        S^{-1}(f) = \frac{S^*(f)}{S^*(f)S(f) + \beta * \epsilon(f)},
```

Comment:
I think we should write AND apply epsilon**2 as noted in Norcross Eq. 2.6. Not sure why we did not square in the earlier implementation - maybe because of the way beta was applied.